utils/lib/pb_type_xml.py

```diff
@@ -90,19 +97,148 @@ def add_switchblock_locations(xml):
     })
 
 
-def start_pb_type(
+def start_sub_tile(sub_tile_name, pin_assignments, input_wires, output_wires):
+    sub_tile_xml = ET.Element(
+        'sub_tile', {
+            'name': add_vpr_tile_prefix(sub_tile_name),
+            'capacity': "1",
+        }
+    )
+
+    # Input definitions for the TILE
+    sub_tile_xml.append(ET.Comment(" Sub Tile Inputs "))
+    for name in sorted(input_wires):
+        input_type = 'input'
+
+        if name.startswith('CLK_BUFG_'):
+            if name.endswith('I0') or name.endswith('I1'):
+                input_type = 'clock'
+        elif 'CLK' in name:
+            input_type = 'clock'
+
+        ET.SubElement(
+            sub_tile_xml,
+            input_type,
+            {
+                'name': name,
+                'num_pins': '1'
+            },
+        )
+
+    # Output definitions for the TILE
+    sub_tile_xml.append(ET.Comment(" Sub Tile Outputs "))
+    for name in sorted(output_wires):
+        ET.SubElement(
+            sub_tile_xml,
+            'output',
+            {
+                'name': name,
+                'num_pins': '1'
+            },
+        )
+
+    return sub_tile_xml
+
+
+def start_tile(
         tile_name,
         pin_assignments,
+        input_wires=None,
+        output_wires=None,
+        sites=None,
+):
+    """ Starts a pb_type by adding input, clock and output tags. """
+    assert bool(input_wires) == bool(output_wires)
```

Comment:
Can you add a comment on what these asserts are checking?

Reply:
This is probably clearer written as an if/else

Reply:
Done